packages/block-library/src/navigation/edit/index.js

```diff
@@ -772,39 +791,42 @@ function Navigation( {
 	return (
 		<EntityProvider kind="postType" type="wp_navigation" id={ ref }>
 			<RecursionProvider uniqueId={ recursionId }>
-				<BlockControls>
-					{ ! isDraftNavigationMenu && isEntityAvailable && (
-						<ToolbarGroup className="wp-block-navigation__toolbar-menu-selector">
-							<NavigationMenuSelector
-								ref={ navigationSelectorRef }
-								currentMenuId={ ref }
-								clientId={ clientId }
-								onSelectNavigationMenu={ ( menuId ) => {
-									handleUpdateMenu( menuId );
+				<InspectorControls>
+					<PanelBody title={ __( 'Menu' ) }>
+						<NavigationMenuSelector
+							ref={ navigationSelectorRef }
+							currentMenuId={ ref }
+							clientId={ clientId }
+							onSelectNavigationMenu={ ( menuId ) => {
+								handleUpdateMenu( menuId );
+								setShouldFocusNavigationSelector( true );
```

Comment:
The code to keep focus on the dropdown doesn't seem to work in this PR. In my testing the `navigationSelectorRef` is `undefined`, but not sure why as the code is much the same.

Reply:
Attempted a fix in [0ede795](https://github.com/WordPress/gutenberg/pull/42987/commits/0ede795475b8b7a5d11055fe813e43226a22fa65)

Reply:
It seems to work the first time switching menus, but after that it doesn't work 🤔 

Here are the steps to repro:
(Using the keyboard)
1. Open the navigation menu selector
2. Switch to another menu
3. Expected: the menu should stay open with the menu item still selected
4. Repeat steps 1-3

Reply:
I think I solved that, may need a rebuild?

Reply:
It's working on my side 🎉